lib/request.js

```diff
@@ -32,175 +33,192 @@ var am = require('../');
-        	/* If we're the top and we're a type, we get a request Id, not an event Id */
-            this.id = nextRequestId;
-            ++nextRequestId;        	
-        }
+function Request(type, name, root) {
+  this.children = [];
+  this.name = name;
+  this.type = type;
+
+  if (root === true) {
+    var reqDomain = domain.create();
+    reqDomain.enter();
+  }
+
+  if (process.domain) {
+    this.parent = process.domain.currentRequest;
+  }
+  this.id = nextEventId;
+  ++nextEventId;
+  if (this.parent) {
+    this.top = this.parent.top;
+    this.parent.children.push(this);
+  } else {
+    this.top = this;
+    if (this.type) {
+      /* If we're the top and we're a type, we get a request Id, not an event Id */
+      this.id = nextRequestId;
+      ++nextRequestId;
     }
+  }
 }
 
 Request.prototype.traceStart = function() {
-    if( !this.tracedStart ) {
-        this.tracedStart = true;
-        // we may delay tracing parent requests until later time when they exceed threshold
-        if( this.parent ) {
-            this.parent.traceStart();
-            if( this.parent.type ) {
-                parentId = this.parent.id;
-            } else {
-                parentId = this.parent.methId;
-            }
-        }
+  if (!this.tracedStart) {
+    this.tracedStart = true;
+    // we may delay tracing parent requests until later time when they exceed threshold
+    if (this.parent) {
+      this.parent.traceStart();
+      // FIXME(sam) assigning to a global parentId that appears to never be
+      // used, but I can't be sure.
+      if (this.parent.type) {
+        parentId = this.parent.id;
+      } else {
+        parentId = this.parent.methId;
+      }
     }
-}
+  }
+};
 
 Request.prototype.traceStop = function() {
-    if( !this.traceStopped ) {
-        this.traceStopped = true;
-        var stack = null;
-
-        if((config.minClockStack != -1) && (this.timer.timeDelta >= config.minClockStack)) {
-            this.stack = this.fetchStack();
-        }
-
-        var parentId;
-        if( this.parent ) {
-            if( this.parent.type ) {
-                parentId = this.parent.id;
-            } else {
-                parentId = this.parent.methId;
-            }
-        }
+  if (!this.traceStopped) {
+    this.traceStopped = true;
+    if (
+      config.minClockStack != -1 &&
+      this.timer.timeDelta >= config.minClockStack
+    ) {
+      this.stack = this.fetchStack();
     }
-}
 
-Request.prototype.start = function() {
-    this.active = true;
-    if( !this.timer ) {
-    	this.timer = timer.start();
-    }
-    if (process.domain) {
-        process.domain.currentRequest = this;
+    // FIXME(sam) assignment to a local parentId which is never used, so either
+    // this whole chunk of code can be deleted, or the `var parentId` is a bug,
+    // and this is supposed to be setting a global, except that wouldn't be
+    // parallel request safe... not sure what to do here.
+    var parentId;
```

Comment:
Here as well, its not assigning to a global, so the entire section is a no-op... which makes me think that it is supposed to be assigning to a global, and that the `var` is a bug... except using a global would be a bug... So, over to you.